src/main/java/org/spongepowered/api/world/World.java

```diff
@@ -157,22 +157,12 @@
     Dimension getDimension();
 
     /**
-     * Gets the random seed for this world.
-     *
-     * @return The seed
-     */
-    long getWorldSeed();
```

Comment:
Yeah, each world has its own WorldGenerator. In this PR the WorldGenerator class holds everything that's needed for world generation, like the chunk generator, the populators and their settings. The way I see it, the seed is just another setting, not more important than which biome generator is active.

Ideally, the world generator would be completely independent of the world. It would just be one big bag of classes and settings. `oneWorld.setWorldGenerator(anotherWorld.getWorldGenerator())` would then make `oneWorld` produce exactly, block for block, the same terrain as `anotherWorld`. In practice, Minecraft's ChunkProviderGenerate keeps a reference to the world object, so it cannot be assigned to another world. (I've made the implementation throw an IllegalArgumentException if you try.)